cmd/restorer/main.go

```diff
@@ -73,38 +70,13 @@ func restore() error {
 		GID:        gid,
 	}
 
-	var cacheStore lifecycle.Cache
-	if cacheImageTag != "" {
-		origCacheImage, err := remote.NewImage(
-			cacheImageTag,
-			auth.EnvKeychain(cmd.EnvRegistryAuth),
-			remote.FromBaseImage(cacheImageTag),
-		)
-		if err != nil {
-			return cmd.FailErr(err, "accessing cache image")
-		}
-
-		emptyImage, err := remote.NewImage(
-			cacheImageTag,
-			auth.EnvKeychain(cmd.EnvRegistryAuth),
-			remote.WithPreviousImage(cacheImageTag),
-		)
-		if err != nil {
-			return cmd.FailErr(err, "creating empty image")
-		}
-
-		cacheStore = cache.NewImageCache(
-			origCacheImage,
-			emptyImage,
-		)
-	} else {
-		var err error
-		cacheStore, err = cache.NewVolumeCache(cacheDir)
-		if err != nil {
-			return cmd.FailErr(err, "create volume cache")
-		}
+	cacheStore, err := cache.MaybeCache(cacheImageTag, cacheDir)
+	if err != nil {
+		return cmd.FailErr(err, "set up cache")
+	}
+	if cacheStore == nil {
+		restorer.Logger.Warn("Not restoring cached layer data, no cache flag specified.")
```

Comment:
Is it even possible to hit this warning given line 53? It seems like we probably want to replace the error on line 54 with this warning and execute `restorer.Restore` regardless, so it can clean up metadata (in the extreme edge case where that is necessary).

Reply:
I completely missed that check. Changed to a warning and removed these lines.